sessionctx/variable/sysvar.go

```diff
@@ -652,28 +657,48 @@ var defaultSysVars = []*SysVar{
 	{Scope: ScopeGlobal, Name: TiDBScatterRegion, Value: BoolToOnOff(DefTiDBScatterRegion), Type: TypeBool},
 	{Scope: ScopeGlobal, Name: TiDBEnableStmtSummary, Value: BoolToOnOff(DefTiDBEnableStmtSummary), Type: TypeBool, AllowEmpty: true,
 		SetGlobal: func(_ context.Context, s *SessionVars, val string) error {
-			return stmtsummary.StmtSummaryByDigestMap.SetEnabled(TiDBOptOn(val))
+			v := TiDBOptOn(val)
+			if s.StmtSummary != nil {
+				s.StmtSummary.SetEnabled(v)
+			}
+			return stmtsummary.StmtSummaryByDigestMap.SetEnabled(v)
```

Comment:
It would be clearer if using `s.StmtSummaryV2` to clarify why set twice.

Reply:
done